### PR TITLE
ws:// processing at Notifier library

### PR DESCRIPTION
**Pull request by fgalan**
Part of #1669 

Unfortunatelly, make ft is broken in feature/1181_websocket due to current WS implementation in that branch, I cannot assure this modification doesn't break anything...

@fortizc

### CODE REVIEW

File: src/lib/ngsiNotify/senderThread.cpp [hunk 1]
```diff
@@ -30,6 +30,18 @@
 #include "ngsiNotify/senderThread.h"
 
 
+// FIXME P11 #1669: remove this stub when the actual sendNotifyContextRequestWs() gets developed a the end. @fortizc please take care of this
+// in the ws library
+int sendNotifyContextRequestWs(const std::string& subId, const std::map<std::string, std::string>& headers, const std::string& data)
```
Comment (fgalan) [inline src/lib/ngsiNotify/senderThread.cpp]: At the begining I proposed (at #1669) to use `bool` as return value for this function. However, in order to align with httpRequestSent() signature used at startSenderThread() function, I think it should be better to use `int` the the usual semantics: 0 -> everythin ok, not 0 -> error.

(I have edited signature at https://github.com/telefonicaid/fiware-orion/issues/1669#issuecomment-199907859 has been changed `void` -> `int`)

File: src/lib/ngsiNotify/Notifier.cpp [hunk 2]
```diff
@@ -104,53 +99,26 @@ void Notifier::sendNotifyContextRequest(NotifyContextRequest* ncr, const std::st
...
-    if (!parseUrl(url, host, port, uriPath, protocol))
-    {
-      LM_E(("Runtime Error (not sending NotifyContextRequest: malformed URL: '%s')", url.c_str()));
-      return;
-    }
-
-    /* Set Content-Type */
-    std::string content_type = "application/json";
-
-#ifdef SEND_BLOCKING
```
Comment (fgalan) [inline src/lib/ngsiNotify/Notifier.cpp]: SEND_BLOCKING has not been in use sinde months (or years?). In addition, keeping this dead code aligned with the changes implemented in this PR for WS support will complicate things, so let's definetively remove it.

File: src/lib/ngsiNotify/Notifier.cpp [hunk 3]
```diff
@@ -36,14 +36,9 @@
...
-/* ****************************************************************************
-*
-* Select 'method' to send notifications - only one can be uncommented
-*/
-//#define SEND_BLOCKING
-#define SEND_IN_NEW_THREAD
-
+// FIME P11 #1669: move this to WS library. @fortizc please take care of this
+#define WS_SCHEME         "ws://"
+#define WS_SCHEME_LENGTH  5
```
Comment (fortizc) [inline src/lib/ngsiNotify/Notifier.cpp]: This is only a suggest, but may be is better use a static const std::string and to get its length you can use size() method, moreover when you assign a char * to std::string internally make something like that: std::string(my_char_pointer) in this way is better too a const std::string

Comment (fgalan) [inline src/lib/ngsiNotify/Notifier.cpp]: Ok. However, after skype talk, this is going to be done by @fortizc in a next PR, so NTC with regards to this PR.

File: src/lib/ngsiNotify/Notifier.cpp (hunk 2 above)
Comment (fgalan) [inline src/lib/ngsiNotify/Notifier.cpp]: NTC

File: src/lib/ngsiNotify/senderThread.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/ngsiNotify/senderThread.cpp]: NTC

**Comment by fortizc** [comment]
LGTM !

**Pull request merged**